src/test.rs

```diff
@@ -126,47 +128,47 @@ mod mpz {
         assert!((&x % -&y).to_string() == (20i32 % -3).to_string());
         assert!((-&x % &y).to_string() == (-20i32 % 3).to_string());
     }
-    
+
     #[test]
     fn test_add() {
-    	let x: Mpz = From::<i64>::from(2);
-    	let y: Mpz = From::<i64>::from(3);
-    	let str5 = 5i32.to_string();
-    	assert!((&x + &y).to_string() == str5);
-    	assert!((&x + 3).to_string() == str5);
-    	assert!((&y + 2).to_string() == str5);
+        let x: Mpz = From::<i64>::from(2);
+        let y: Mpz = From::<i64>::from(3);
+        let str5 = 5i32.to_string();
+        assert!((&x + &y).to_string() == str5);
+        assert!((&x + 3).to_string() == str5);
+        assert!((&y + 2).to_string() == str5);
     }
-    
+
     #[test]
     fn test_sub() {
-    	let x: Mpz = From::<i64>::from(2);
-    	let y: Mpz = From::<i64>::from(3);
-    	assert!((&x - &y).to_string() == (-1i32).to_string());
-    	assert!((&y - &x).to_string() == 1i32.to_string());
-    	assert!((&y - 8).to_string() == (-5i32).to_string());
+        let x: Mpz = From::<i64>::from(2);
+        let y: Mpz = From::<i64>::from(3);
+        assert!((&x - &y).to_string() == (-1i32).to_string());
+        assert!((&y - &x).to_string() == 1i32.to_string());
+        assert!((&y - 8).to_string() == (-5i32).to_string());
     }
-    
+
     #[test]
     fn test_mul() {
-    	let x: Mpz = From::<i64>::from(2);
-    	let y: Mpz = From::<i64>::from(3);
-    	assert!((&x * &y).to_string() == 6i32.to_string());
-    	assert!((&x * 3i64).to_string() == 6i32.to_string());
-    	assert!((&y * -5i64).to_string() == (-15i32).to_string());
-    	// check with values not fitting in 32 bits
-    	assert!((&x * 5000000000i64).to_string() == 10000000000i64.to_string());
+        let x: Mpz = From::<i64>::from(2);
+        let y: Mpz = From::<i64>::from(3);
+        assert!((&x * &y).to_string() == 6i32.to_string());
+        assert!((&x * 3i64).to_string() == 6i32.to_string());
+        assert!((&y * -5i64).to_string() == (-15i32).to_string());
+        // check with values not fitting in 32 bits
+        assert!((&x * 5000000000i64).to_string() == 10000000000i64.to_string());
     }
 
     #[test]
     fn test_to_str_radix() {
         let x: Mpz = From::<i64>::from(255);
-        assert!(x.to_str_radix(16) == "ff".to_string());
+        assert!(x.to_str_radix(16) == *"ff");
```

Comment:
The `*` isn’t needed; simply `assert!(x.to_str_radix(16) == "ff")` works fine (or `assert_eq!`).

Reply:
Cool, this surprised me, I didn't think `assert_eq!` could compare an `&str` with a `String`. Thanks, implemented as suggested. I added it as a new patch on top of this branch to make re-review easier. I can squash it with the original one if you prefer.

Reply:
Rust has a boatload of [`PartialEq` impls](https://doc.rust-lang.org/std/cmp/trait.PartialEq.html#implementors), including `PartialEq<str> for str`, `PartialEq<String> for String`, `PartialEq<str> for String`, `PartialEq<String> for str`, `PartialEq<&'a str> for String`, `PartialEq<String> for &'a str`.

You haven’t actually pushed anything though. I’m not the maintainer, but for my part I’m fine with squashing.

Reply:
Squashed and force pushed. Thanks for the tip about `PartialEq` implementations!